File: celery-worker/src/clients/queries.py
# GraphQL queries for journal service

GET_JOURNAL_ENTRY_BY_ID = """
query GetJournalEntry($entryId: UUID!) {
    journalEntry(entryId: $entryId) {
        __typename
        id
        userId
        entryType
        createdAt
        modifiedAt
        ... on FreeformJournalEntry {
            content: payload
        }
        ... on GratitudeJournalEntry {
            payload {
                gratefulFor
                excitedAbout
                focus
                affirmation
                mood
            }
        }
        ... on ReflectionJournalEntry {
            payload {
                wins
                improvements
                mood
            }
        }
    }
}
"""

GET_JOURNAL_ENTRIES = """
query GetJournalEntries($limit: Int) {
    journalEntries(limit: $limit) {
        __typename
        id
        userId
        entryType
        createdAt
        modifiedAt
        ... on FreeformJournalEntry {
            content: payload
        }
        ... on GratitudeJournalEntry {
            payload {
                gratefulFor
                excitedAbout
                focus
                affirmation
                mood
            }
        }
        ... on ReflectionJournalEntry {
            payload {
                wins
                improvements
                mood
            }
        }
    }
}
"""

# Additional queries for future use
GET_JOURNAL_ENTRIES_BY_USER = """
query GetJournalEntriesByUser($userId: UUID!, $limit: Int) {
    journalEntries(userId: $userId, limit: $limit) {
        __typename
        id
        userId
        entryType
        createdAt
        modifiedAt
        ... on FreeformJournalEntry {
            content: payload
        }
        ... on GratitudeJournalEntry {
            payload {
                gratefulFor
                excitedAbout
                focus
                affirmation
                mood
            }
        }
        ... on ReflectionJournalEntry {
            payload {
                wins
                improvements
                mood
            }
        }
    }
}
"""

GET_JOURNAL_ENTRIES_BY_DATE_RANGE = """
query GetJournalEntriesByDateRange($startDate: DateTime!, $endDate: DateTime!, $limit: Int) {
    journalEntries(startDate: $startDate, endDate: $endDate, limit: $limit) {
        __typename
        id
        userId
        entryType
        createdAt
        modifiedAt
        ... on FreeformJournalEntry {
            content: payload
        }
        ... on GratitudeJournalEntry {
            payload {
                gratefulFor
                excitedAbout
                focus
                affirmation
                mood
            }
        }
        ... on ReflectionJournalEntry {
            payload {
                wins
                improvements
                mood
            }
        }
    }
}
"""

File: mindmirror-mobile/src/services/api/queries.ts
import { gql } from '@apollo/client'

// Tradition-related queries
export const LIST_TRADITIONS = gql`
  query ListTraditions {
    listTraditions
  }
`

// Journal entry queries
export const JOURNAL_ENTRY_EXISTS_TODAY = gql`
  query JournalEntryExistsToday($entryType: String!) {
    journalEntryExistsToday(entryType: $entryType)
  }
`

export const GET_JOURNAL_ENTRIES = gql`
  query GetJournalEntries {
    journalEntries {
      __typename
      ... on GratitudeJournalEntry {
        id
        createdAt
        payload {
          gratefulFor
          excitedAbout
          focus
          affirmation
          mood
        }
      }
      ... on ReflectionJournalEntry {
        id
        createdAt
        payload {
          wins
          improvements
          mood
        }
      }
      ... on FreeformJournalEntry {
        id
        createdAt
        content: payload
      }
    }
  }
`

// Chat-related query
export const ASK_QUERY = gql`
  query Ask($query: String!, $tradition: String!, $includeJournalContext: Boolean) {
    ask(query: $query, tradition: $tradition, includeJournalContext: $includeJournalContext)
  }
`

// Meal suggestion query
export const GET_MEAL_SUGGESTION = gql`
  query GetMealSuggestion($mealType: String!, $tradition: String!) {
    getMealSuggestion(mealType: $mealType, tradition: $tradition) {
      suggestion
      reasoning
    }
  }
`

// Journal summarization query
export const SUMMARIZE_JOURNALS_QUERY = gql`
  query SummarizeJournals {
    summarizeJournals {
      summary
      generatedAt
    }
  }
`